test/role.test.js

```diff
@@ -431,6 +431,261 @@ describe('role model', function() {
+      var {sender, writer, messages} = data;
+      var writerRole = {
+        principalType: ACL.USER, principalId: writer.id,
+        model: Message, id: messages[0].id,
+      };
+      var senderRole = {
+        principalType: ACL.USER, principalId: sender.id,
+        model: Message, id: messages[1].id,
+      };
+      var noOwnerRole = {
+        principalType: ACL.USER, principalId: sender.id,
+        model: Message, id: messages[2].id,
+      };
+
+      return Promise.all([
+        new Promise((resolve, reject) => {
+          Role.isInRole(Role.OWNER, writerRole, (err, yes) => {
+            if (err) return reject(err);
+            resolve(!yes);
+          })
+        }),
+        new Promise((resolve, reject) => {
+          Role.isInRole(Role.OWNER, senderRole, (err, yes) => {
+            if (err) return reject(err);
+            resolve(yes);
+          });
+        }),
+        new Promise((resolve, reject) => {
+          Role.isInRole(Role.OWNER, noOwnerRole, (err, yes) => {
+            if (err) return reject(err);
+            resolve(!yes);
+          });
+        }),
+      ]);
+    }).catch(done)
+      .map(bool => assert(bool))
+      .then(_ => done());
+  });
+
+  it('should resolve owner when no flag on userId property', function(done) {
+    var Message = app.registry.createModel('message',
+    { content: String, userId: Number },
+    {
+      relations: {
+        // It shouldn't resolve the `sender` role
+        sender: {
+          type: 'belongsTo',
+          model: 'User',
+          foreignKey: 'senderId',
+        },
+      }
+    });
+
+    app.model(Message, {dataSource: 'db'});
+
+    var password = 'pass';
+    new Promise((resolve, reject) => {
+      app.models.User.create(
+        [
+          {email: 'user@user.com', password},
+          {email: 'sender@sender.com', password}
+        ], (err, users) => {
+          if (err) return reject(err);
+          resolve(users);
+        });
+    }).then(users => {
+      var user = users[0];
+      var sender = users[1];
+      var userId = user.id;
+      var senderId = sender.id;
+      return Promise.props({
+        sender, user,
+        messages: new Promise((resolve, reject) => {
+          Message.create([
+            {userId}, {userId, senderId}, {}
+          ], (err, messages) => {
+            if (err) return reject(err);
+            resolve(messages);
+          })
+        })
+      });
+    }).then(data => {
+      var {sender, user, messages} = data;
+      var userRole = {
+        principalType: ACL.USER, principalId: user.id,
+        model: Message, id: messages[0].id,
+      };
+      var senderRole = {
+        principalType: ACL.USER, principalId: user.id,
+        model: Message, id: messages[1].id,
+      };
+      var noOwnerRole = {
+        principalType: ACL.USER, principalId: sender.id,
+        model: Message, id: messages[2].id,
+      };
+
+      return Promise.all([
+        new Promise((resolve, reject) => {
```

Comment:
We try to keep pull request small and have each request focused on a single thing. Adding Promise support AND allowing multi-owner Role resolution in the same pull request would go against that.

So I see two options how to move this pull request forward:
 - Rewrite your new tests to use callbacks instead of promises. This seems like a step back to me.
 - Start a new feature branch & a new pull request that will only add support for Promise style invocation to `Role` methods (and possibly other models you are going to invoke the tests you are adding here). When this other pull request is landed, you can simplify the code in this pull request by using the newly added Promise API.